lib/auth/init_test.go

```diff
@@ -141,7 +140,7 @@ func TestBadIdentity(t *testing.T) {
 
 	// bad cert type
 	_, err = state.ReadSSHIdentityFromKeyPair(priv, pub)
-	require.IsType(t, trace.BadParameter(""), err)
+	require.ErrorIs(t, trace.BadParameter("failed to parse server certificate: not an SSH certificate"), err)
```

Comment:
We don't care about capturing a trace here, and `errors.Is` takes the error first and then the comparison target.

```suggestion
	require.ErrorIs(t, err, &trace.BadParameterError{Message: "failed to parse server certificate: not an SSH certificate"})
```

That being said, this feels like a very specific check and we should probably relax it a bit like


```suggestion
	require.ErrorAs(t, err, new(*trace.BadParameterError))
	require.ErrorContains(t, err, "not an SSH certificate")
```

That being said, these changes seem a bit spurious and could be reverted here and addressed in a separate PR.